the-collab-lab/tcl-11-smart-shopping-list
# Feature(Issue #4) : Add a new item to a shopping list to start recording purchases

## Description



This PR adds on to the functionality of our `<AddItem />` component, presenting the user with a form that allows them to enter the item name, as well as an estimate of when they'd want to purchase said item once more. Also added is a data point that shows the "Last purchased date", but as of now, it just defaults to null.

## Related Issue
closes #4 

## Acceptance Criteria

- User is presented with a form that allows them to enter the name of the item and select how soon they anticipate needing to buy it again (Soon, Kind of Soon, Not Soon)
- When the user submits the form, the item is saved to the database, associated with the user’s token
- Along with the item name, an integer corresponding to the estimated number of days until next purchase is saved: 7 for “Soon”, 14 for “Kind of Soon”, and 30 for “Not Soon”
- The last purchased date should be set to null initially (i.e. you can create an item without purchasing it)

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
| ✓   | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before
![image](https://user-images.githubusercontent.com/6282435/90321892-f114ba80-df1b-11ea-848e-e84b872db7d6.png)




### After

![image](https://user-images.githubusercontent.com/6282435/90321902-0558b780-df1c-11ea-8438-d5f49159d91c.png)


## Testing Steps / QA Criteria

[Screencast showing Testing Steps](https://www.youtube.com/watch?v=B1l_4EFeyrk&feature=youtu.be)

- From your terminal, run `git checkout bl-sd-add-item-to-shopping-list`
- If this is your first time running this repo, don't forget to install dependencies: `npm install`
- Once you're done preparing your environment, run `npm run start` then go to `localhost:3000`
- Click the **List** navigation link seen on the bottom of the homepage.
- Click the "add item" button to create a new shopping list as seen in the screencast above
- After being redirected to the `<AddItem />` component, add item using the form, and set the repurchase period that you expect.  When your item is filled out, click the big "Add Item" button.
- Return to the <List /> view using the navigation link once more. The item you added should show up.
- If you have firebase access, go to firebase's database view and verify that additional datapoints (Resupply Period and Last purchased date) are added, using your token found in `localstorage` to verify.
